xtable-api/src/main/java/org/apache/xtable/model/schema/InternalPartitionField.java

```diff
@@ -47,4 +49,6 @@ public class InternalPartitionField {
   @Builder.Default List<String> partitionFieldNames = Collections.emptyList();
   // An enum describing how the source data was transformed into the partition value
   PartitionTransformType transformType;
+  // Transform options such as number of buckets in the BUCKET transform type
+  @Builder.Default Map<String, Object> transformOptions = Collections.emptyMap();
```

Comment:
Instead of Object, should we add a new class known as `InternalValue` ? 
```
InternalValue { 
  InternalType type; 
  Object value;
}
```

Reply:
Right now, I made this similar to the `metadata` in `InternalSchema`. Another option I considered is to change `PartitionTransformType` into a class which can then have its own instance variables.